dashboard/src/shared/types.ts

```diff
@@ -383,6 +383,10 @@ export interface IBasicFormParam {
   title?: string;
   minimum?: number;
   maximum?: number;
+  render?: string;
+  sliderMin?: number;
+  sliderMax?: number;
+  sliderUnit?: string;
```

Comment:
So the awkwardness is that we have a generic param interface which is having fields added for specifics. Is it possible instead to compose a new `IBasicFormNumberWithUnitsStringParam` which includes `IBasicFormParam` as well as `minimum`, `maximum`, `units'?

My head isn't in this code deeply enough to see if that would avoid the awkwardness or make it more convoluted - so just see what you think.

Reply:
that would be more difficult to handle from the code point of view. In that case, the type of the parameters will be `IBasicFormParam | IBasicFormNumberWithUnitsStringParam`, which make it more difficult because it require type conversions and assumptions. IMO, it's easier to handle as optional parameters.